cmd/ontap-cvo.go

```diff
@@ -652,6 +664,33 @@ func formatSharesMap(shares map[string][]string) map[string]string {
 	return result
 }
 
+// This will check for duplicate shares
+func validateUserInput(client *kubernetes.Clientset, namespace string, svm string, shareToCheck string) error {
+	// Retrieve the users existing-cm
+	klog.Infof("Validating that user did not submit duplicate request " + namespace)
+	existingSharesCM, err := client.CoreV1().ConfigMaps(namespace).Get(context.Background(), existingSharesConfigMapName, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		klog.Infof("User does not have an existing-shares configmap.")
+		return nil
+	}
+	userSharesData, err := unmarshalSharesMap(existingSharesCM.Data)
+	if err != nil {
+		return fmt.Errorf("error while unmarshalling existing shares configmap in %s: %v", namespace, err)
+	}
+	for share := range userSharesData[svm] {
+		if userSharesData[svm][share] == shareToCheck {
+			klog.Errorf("Duplicate share:%s already exists for svm:%s in namespace: %s", shareToCheck, svm, namespace)
+			return &ShareError{
```

Comment:
Here, you should just return fmt.Errorf(). The ShareError is made wherever this function is called (like you already did).